confmap/provider/internal/configurablehttpprovider/provider.go

```diff
@@ -96,7 +96,7 @@ func (fmp *provider) Retrieve(_ context.Context, uri string, _ confmap.WatcherFu
 	if err != nil {
 		return nil, fmt.Errorf("unable to download the file via HTTP GET for uri %q: %w ", uri, err)
 	}
-	defer resp.Body.Close()
+	defer func() { _ = resp.Body.Close() }()
```

Comment:
Does this impact performance on the critical path for no good reasons? I would retain from doing it unless a good reason to do so.

Reply:
Same everywhere in no-test code.

Reply:
I was curious, so created a benchmark to compare the two behaviors:

and in this simple example, explicitly ignoring the error in a closure is actually more performant (now i want to know why that is):

https://gist.github.com/JohnRoesler/d3bf89d60630bc60e6afd8d14ab88e90